src/azul/service/responseobjects/elastic_request_builder.py

```diff
@@ -253,18 +254,24 @@ def open_and_return_json(file_path):
             file_.close()
         return loaded_file
 
-    @staticmethod
-    def apply_paging(es_search, pagination):
+    def apply_paging(self, es_search, pagination, entity_type):
         """
         Applies the pagination to the ES Search object
         :param es_search: The ES Search object
         :param pagination: Dictionary with raw entries from the GET Request.
         It has: 'size', 'sort', 'order', and one of 'search_after', 'search_before', or 'from'.
+        :param entity_type: the string referring to the entity type used to get
+        the ElasticSearch mappings
         :return: An ES Search object where pagination has been applied
         """
-        # Extract the fields for readability (and slight manipulation)
+        # Find the data type of the sort parameter
+        index_client = elasticsearch.client.IndicesClient(self.es_client)
+        index_name = config.es_index_name(entity_type)
+        field_map = index_client.get_field_mapping(fields=[pagination['sort']], index=index_name)
```

Comment:
It would be sad if we had to do an additional ES round-trip for every page. Let's talk about alternatives tomorrow.